tests/test_pegen.py

```diff
@@ -637,13 +637,25 @@ def test_unreachable_implicit1() -> None:
 
 def test_unreachable_implicit2() -> None:
     source = """
-    start: NAME | invalid_input
+    start: NAME | '(' invalid_input ')'
     invalid_input: NUMBER { None }
     """
     grammar = parse_string(source, GrammarParser)
     out = io.StringIO()
     genr = PythonParserGenerator(grammar, out, unreachable_formatting="This is a test")
     genr.generate("<string>")
+    assert "This is a test" in out.getvalue()
+
+
+def test_unreachable_implicit3() -> None:
```

Comment:
I am assuming thsi tests covers all the weird things you saw that made you make this pr, but if there are more, please make a new PR with more tests :)

Reply:
This covers the cases that are actually used in the python grammar. I did not try to build contrived examples to go through all the branches of the visitor through. I will also check that my wording of the documentation is accurate.